### PR TITLE
Adds pinnings for ffmpeg, libblitz, libmatio, libsvm, sox, vlfeat.

### PR DESCRIPTION
**Pull request by 183amir**
For `libblitz`: http://upstream.rosalinux.ru/versions/blitz.html

For `ffmpeg`: http://abi-laboratory.pro/tracker/timeline/ffmpeg/

For `vlfeat`: They keep adding new features in patch version: http://www.vlfeat.org/ and major version is `0`. If they are following the semantic versioning schemes: `Major version zero (0.y.z) is for initial development. Anything may change at any time. The public API should not be considered stable.`. New features will break our packages: https://github.com/conda-forge/staged-recipes/issues/475 but however maybe pinning like `>0.9.20` be a better option but since this is version `0` and they don't follow good practices, the safest bet is to pin it on a patch version. They don't release often.

For `sox`: There is no guarantee that they will not break API: http://upstream.rosalinux.ru/versions/libsox.html

I am guessing API compatibility for `libmatio`, `libsvm`.

### CODE REVIEW

**Comment by 183amir** [comment]
These are the dependencies that I care about and I don't want them to break my packages in future.
There maybe many more packages that we need to pin.

**Comment by 183amir** [comment]
I read their change log and guessed.

**Comment by 183amir** [comment]
Also: http://semver.org/#spec-item-4 `Major version zero (0.y.z) is for initial development. Anything may change at any time. The public API should not be considered stable.`

**Comment by patricksnape** [comment]
True, but that assumes 1) That they are following semver 2) That they would some day move to 1.x

It doesn't quite work if you stay 0.x forever!

**Comment by 183amir** [comment]
Yes but I also their changelog also suggests that they add new features in patch version:
```
News

14/1/2015 VLFeat 0.9.20 released
    Maintenance release. Bugfixes.
12/9/1014 MatConvNet
    Looking for an easy-to-use package to work with deep convolutional neural networks in MATLAB? Check out our new MatConvNet toolbox!
12/9/2014 VLFeat 0.9.19 released
    Maintenance release. Minor bugfixes and fixes compilation with MATLAB 2014a.
29/01/2014 VLFeat 0.9.18 released
    Several bugfixes. Improved documentation, particularly of the covariant detectors. Minor enhancements of the Fisher vectors. [Details] 
22/06/2013 VLFeat 0.9.17 released
    Rewritten SVM implementation, adding support for SGD and SDCA optimizers and various loss functions (hinge, squared hinge, logistic, etc.) and improving the interface. Added infrastructure to support multi-core computations using OpenMP. Added OpenMP support to KD-trees and KMeans. Added new Gaussian Mixture Models, VLAD encoding, and Fisher Vector encodings (also with OpenMP support). Added LIOP feature descriptors. Added new object category recognition example code, supporting several standard benchmarks off-the-shelf. This is the third point update supported by the PASCAL Harvest programme. [Details] 
01/10/2012 VLBenchmarks 1.0-beta released.
    This new project provides simple to use benchmarking code for feature detectors and descriptors. Its development was supported by the PASCAL Harvest programme. [Details]
01/10/2012 VLFeat 0.9.16 released
    Added VL_COVDET() (covariant feature detector). This function implements the following detectors: DoG, Hessian, Harris Laplace, Hessian Laplace, Multiscale Hessian, Multiscale Harris. It also implements affine adaptation, estimation of feature orientation, computation of descriptors on the affine patches (including raw patches), and sourcing of custom feature frame. Added the auxiliary function VL_PLOTSS(). This is the second point update supported by the PASCAL Harvest programme. [Details]
11/9/2012 VLFeat 0.9.15 released
    Added VL_HOG() (HOG features). Added VL_SVMPEGASOS() and a vastly improved SVM implementation. Added IHASHSUM (hashed counting). Improved INTHIST (integral histogram). Added VL_CUMMAX(). Improved the implementation of VL_ROC() and VL_PR(). Added VL_DET() (Detection Error Trade-off (DET) curves). Improved the verbosity control to AIB. Added support for Xcode 4.3, improved support for past and future Xcode versions. Completed the migration of the old test code in toolbox/test, moving the functionality to the new unit tests toolbox/xtest. Improved credits. This is the first point update supported by the PASCAL Harvest (several more to come shortly). A big thank to our sponsor! [Details]. 
10/1/2012 PASCAL2 Harvest funding
    In the upcoming months many new functionalities will be added to VLFeat thanks to the PASCAL Harvest! See here for details. 

 
© 2007-13 The authors of VLFeat 
```

**Comment by patricksnape** [comment]
Sure, but new features aren't breaking changes. Only changing previously provided public API behaviour/existence is. Anyway, I agreed with you in that their version system is merely one of their own and so pinning it seems the safest option!

**Comment by 183amir** [comment]
New features will break our packages: https://github.com/conda-forge/staged-recipes/issues/475
but however maybe pinning like `>0.9.20` be a better option but since this is version `0` and they don't follow good practices I want to be sure.

**Comment by patricksnape** [comment]
As I said, I think pinning this is safe since vlfeat seems to be following no real versioning scheme whatsoever (the first version was `0.9.15`?). I 👍 the vlfeat pinning!

However, without trying to derail the conversation I would say that versioning is hard and it really depends on the semantics of your version numbers. Under 100% strict semver versioning a patch number should *only* indicate a backwards compatible bug fix and if any new functionality or breaking changes are added then the minor or major number should be bumped respectively. If your software is sensitive to a specific public API then you should be sensitive up to the minor number. Else you should be sensitive up to the major. Then, to make it even more fun, everyone just uses version `0.x` (because versioning is hard) so that they can actually do whatever they like 😆!

File: scripts/pin_the_slow_way.py [hunk 1]
```diff
@@ -31,19 +31,25 @@
 pinned = {
           'boost': 'boost 1.61.*',
           'bzip2': 'bzip2 1.0.*',
+          'ffmpeg': 'ffmpeg 2.8.*',
```
Comment (jakirkham) [inline scripts/pin_the_slow_way.py]: 👍

File: scripts/pin_the_slow_way.py [hunk 2]
```diff
@@ -31,19 +31,25 @@
...
           'fontconfig': 'fontconfig 2.11.*',
           'freetype': 'freetype 2.6.*',
           'hdf5': 'hdf5 1.8.17|1.8.17.*',
           'icu': 'icu 56.*',
           'jpeg': 'jpeg 9*',
+          'libblitz': 'libblitz 0.10|0.10.*',
+          'libmatio': 'libmatio 1.5.*',
           'libnetcdf': 'libnetcdf 4.4.*',
-          'libpng': 'libpng >=1.6.21,<1.7',
+          'libpng': 'libpng 1.6.23',
```
Comment (jakirkham) [inline scripts/pin_the_slow_way.py]: This is being done in PR ( https://github.com/conda-forge/conda-forge.github.io/pull/175 ).

**Comment by jakirkham** [comment]
Happy with the `ffmpeg` pinning. The `libpng` pinning is being addressed in another PR. Not really familiar with the rest so will trust your judgement.

**Comment by 183amir** [comment]
LGTM?

**Comment by jakirkham** [comment]
@conda-forge/core, any comments by anyone to these pinnings? It seems we are in agreement. Would it be too extreme to say we should merge this within 24hrs if there are no objections?

**Comment by patricksnape** [comment]
All good from my end.

**Comment by jjhelmus** [comment]
LGTM

**Comment by 183amir** [comment]
Could you please merge this then?

**Comment by 183amir** [comment]
Although this has been merged, I have not seen the pull requests suggesting these pins on my feedstocks. Am I missing something here? Is our automatic pin process broken still?
ping @conda-forge/core

**Comment by lvc** [comment]
Hello,

Missed libraries have been added to the tracker:

http://abi-laboratory.pro/tracker/timeline/vlfeat/
http://abi-laboratory.pro/tracker/timeline/matio/
http://abi-laboratory.pro/tracker/timeline/libsvm/

**Comment by jakirkham** [comment]
@183amir, could you please open a new issue on this repo with a list of the feedstocks that have not received pinning PRs?

**Comment by jakirkham** [comment]
Thanks @lvc. That's very nice of you.

**Comment by 183amir** [comment]
@jakirkham I thought the whole purpose of adding the pinnings here was that they would be fixed automatically. Why do you need a list of repos for these pull requests?

**Comment by jakirkham** [comment]
Because the pinning service is working on repos. So if you are having issues, these are special case failures.

In any event, the real point is that we should no longer be discussing this in a merged PR. This is going to get lost. Please open it in a new issue with this info, @183amir.

**Comment by 183amir** [comment]
@lvc could you add `giflib`(https://sourceforge.net/projects/giflib) to this website: http://abi-laboratory.pro/tracker/ too, please?

**Comment by lvc** [comment]
Done: http://abi-laboratory.pro/tracker/timeline/giflib/

Feel free to add more libraries to the tracker (add an issue to https://github.com/lvc/upstream-tracker/issues). I'm interested to add all "live" and popular libraries.

Thank you.

**Comment by 183amir** [comment]
Thank you @lvc

File: scripts/pin_the_slow_way.py [hunk 3]
```diff
@@ -31,19 +31,25 @@
...
           'boost': 'boost 1.61.*',
           'bzip2': 'bzip2 1.0.*',
+          'ffmpeg': 'ffmpeg 2.8.*',
           'fontconfig': 'fontconfig 2.11.*',
           'freetype': 'freetype 2.6.*',
           'hdf5': 'hdf5 1.8.17|1.8.17.*',
           'icu': 'icu 56.*',
           'jpeg': 'jpeg 9*',
+          'libblitz': 'libblitz 0.10|0.10.*',
+          'libmatio': 'libmatio 1.5.*',
```
Comment (jakirkham) [inline scripts/pin_the_slow_way.py]: Looks like `libmatio` might need to be pinned tighter than 1.5.*.

ref: https://abi-laboratory.pro/tracker/timeline/matio/

**Pull request merged**